client/electron/go_plugin.ts

```diff
@@ -36,7 +34,85 @@ export async function invokeMethod(
   method: string,
   input: string
 ): Promise<string> {
-  if (!invokeMethodFunc) {
+  console.debug(`[Backend] - calling InvokeMethod "${method}" ...`);
+  const result = await ensureCgo().invokeMethod(method, input);
+  console.debug(`[Backend] - InvokeMethod "${method}" returned`, result);
+  if (result.ErrorJson) {
+    throw Error(result.ErrorJson);
+  }
+  return result.Output;
+}
+
+/**
+ * Represents a callback function for an event.
+ * @param data The event data string passed from the source.
+ * @param param The param string provided during registration.
+ */
+export type CallbackFunction = (data: string, param: string) => void;
+
+/**
+ * Subscribes to an event from the Go backend.
+ *
+ * @param name The name of the event to subscribe to.
+ * @param callback The callback function to be called when the event is fired.
+ * @param param An optional parameter to be passed to the callback function.
+ * @returns A Promise that resolves when the subscription is successful.
+ *
+ * @remarks Subscribing to an event will replace any previously subscribed callback for that event.
+ */
+export async function subscribeEvent(
```

Comment:
My understanding is that this will become a regular MethodChannel call that takes a callback id?

Reply:
Right, done.